### PR TITLE
feat(abigen): add `--contract` flag to specify a given contract

### PR DESCRIPTION
**Pull request by mask-pp**
## 1. Purpose or design rationale of this PR
i.g: `abigen --contract L1ScrollMessenger` cmd it just translate L1ScrollMessenger contract.
This flag can help to fix this kind of [error](http://jenkins.scroll.tech:8080/blue/organizations/jenkins/scroll-tests%2Fscroll/detail/scroll/5169/pipeline).
...

## 2. Deployment tag versioning

Has the version in `params/version.go` been updated?

- [ ] This PR doesn't involve a new deployment, git tag, docker image tag, and it doesn't affect traces
- [ ] Yes

### CODE REVIEW

**Review by NazariiDenha — DISMISSED** [review]
LGTM
One question, why do you use `c.IsSet && c.String` instead of `c.GlobalIsSet && c.GlobalString`

**Comment by mask-pp** [comment]
> LGTM One question, why do you use `c.IsSet && c.String` instead of `c.GlobalIsSet && c.GlobalString`

Great, we should use `c.GlobalIsSet && c.GlobalString`.

**Comment by mask-pp** [comment]
> > LGTM One question, why do you use `c.IsSet && c.String` instead of `c.GlobalIsSet && c.GlobalString`
> 
> Great, we should use `c.GlobalIsSet && c.GlobalString`.

And fixed.

File: cmd/abigen/main.go [hunk 1]
```diff
@@ -225,6 +230,13 @@ func abigen(c *cli.Context) error {
 		}
 		// Gather all non-excluded contract for binding
 		for name, contract := range contracts {
+			// fully qualified name is of the form <solFilePath>:<type>
+			nameParts := strings.Split(name, ":")
+			typeName := nameParts[len(nameParts)-1]
+			// If contract flag is used and then choose the specified.
+			if c.GlobalIsSet(contractFlag.Name) && c.GlobalString(contractFlag.Name) != typeName {
```
Comment (Thegaram) [inline cmd/abigen/main.go]:
```suggestion
			// The fully qualified name is of the form <solFilePath>:<type>
			nameParts := strings.Split(name, ":")
			typeName := nameParts[len(nameParts)-1]
			// If a contract name is provided then ignore all other contracts
			if c.GlobalIsSet(contractFlag.Name) && c.GlobalString(contractFlag.Name) != typeName {
```

File: cmd/abigen/main.go [hunk 2]
```diff
@@ -99,6 +99,10 @@ var (
 		Name:  "alias",
 		Usage: "Comma separated aliases for function and event renaming, e.g. original1=alias1, original2=alias2",
 	}
+	contractFlag = cli.StringFlag{
+		Name:  "contract",
+		Usage: "Use this flag can just translate specified contract",
```
Comment (Thegaram) [inline cmd/abigen/main.go]:
```suggestion
		Usage: "Name of the contract to generate the bindings for",
```

**Review by Thegaram — DISMISSED** [review]

**Comment by 0xmountaintop** [comment]
looks good after fixing Peter's comments

**Review by 0xmountaintop — APPROVED** [review]

**Review by zimpha — APPROVED** [review]

**Review by Thegaram — APPROVED** [review]

**Pull request merged**